.gitignore

```diff
@@ -1,3 +1,18 @@
+# ignore everything
+*
+
+# unignore all with extension
+!*.*
+
+# unignore all directories
+!*/
+
+# unignore all Makefile
+!*/Makefile
+
+# unignore all Dockerfile
+!*/Dockerfile
+
```

Comment:
@levincent06 Actually no, I forgot that we can't remove these because if we do that then it tries to add all of the executables as well. This is why we should probably put all of our executables into a separate `build` folder or something so that we can just ignore that folder and not worry about the other stuff. But that can be another issue.....